pkg/kubevirt/client.go

```diff
@@ -318,6 +347,43 @@ func (c *client) GetDataVolume(ctx context.Context, namespace string, name strin
 	return dv, nil
 }
 
+func (c *client) GetPersistentVolumeClaim(ctx context.Context, namespace string, claimName string) (*k8sv1.PersistentVolumeClaim, error) {
+	pvc, err := c.infraKubernetesClient.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, claimName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Error getting volume claim %s in namespace %s: %v", claimName, namespace, err)
+		return nil, err
+	}
+	if pvc != nil {
+		if !containsLabels(pvc.Labels, c.infraLabelMap) || !strings.HasPrefix(pvc.GetName(), c.volumePrefix) {
+			return nil, ErrInvalidVolume
+		}
+	}
+	return pvc, nil
+}
+
+func (c *client) ExpandPersistentVolumeClaim(ctx context.Context, namespace string, claimName string, desiredSize int64) error {
+	currentPVC, err := c.GetPersistentVolumeClaim(ctx, namespace, claimName)
+	if err != nil {
+		return err
+	}
+	desiredQuantity := *resource.NewQuantity(desiredSize, resource.DecimalSI)
+	currentQuantity := currentPVC.Spec.Resources.Requests.Storage()
+	if currentQuantity.Cmp(desiredQuantity) >= 0 {
+		if currentQuantity != nil {
```

Comment:
This indicates that currentQuantity can be nil, but we just did a cmp on a potentially nil value? Maybe rewrite it like this?
```go
if currentQuantity != nil && currentQuantity.Cmp(desiredQuantity) >= 0 {
  klog.V(5).Infof("Volume %s of quantity %v is larger than requested quantity %v, no need to expand", claimName, *currentQuantity, desiredQuantity)
  return nil
} else if currentQuantity == nil {
  return nil
}
```

Reply:
My bad, that nil check was redundant